zipline/finance/slippage.py

```diff
@@ -95,7 +107,9 @@ def __repr__(self):
                    volume_limit=self.volume_limit,
                    price_impact=self.price_impact)
 
-    def process_order(self, bar_data, price, volume, order):
+    def process_order(self, data, order):
+        price = data.current(order.asset, "price")
```

Comment:
I would advise only every accessing price after the volume check. (And we should add a comment that says 'do not move this before the `cur_volume`'.)

If there is no volume, i.e. when an asset is illiquid, the 'get last traded value' branch of 'price' will be triggered.

When processing orders, the value used should only be the current close.
I would also recommend using `close` instead of `price` since the ffill behavior of `price` should not be used in this context.